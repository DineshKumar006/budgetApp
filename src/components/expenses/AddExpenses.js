import React,{Component} from 'react';
import * as ExpensesActions from '../../ReduxStore/actions/ExpenseActions';
import {connect} from 'react-redux'
import Form from '../form/form'
import moment from 'moment';


export class AddExpenses extends Component{


    submitExpenses=(value)=>{
       const {amount,description,createdAt,note}=value
        this.props.addExpenses(amount,description,createdAt,note)
        this.props.history.push('/')
    }

    render(){
        return(
            <div>
                <p>Add expenses page</p>
                <hr></hr>
                <label>addExpenses</label>
            
                <Form submitExpenses={(value)=>this.submitExpenses(value)}/>

            </div>
        )
    }
}


const mapStateToProps=(state)=>{
    return{
        expensesData:state.expenseReducer,
        filterData:state.filterReducer
    }

}

const mapDispatchToProps=(dispatch)=>{
    return{
        addExpenses:(amount,description,createdAt,note)=>dispatch(ExpensesActions.addExpenses({amount,description,createdAt,note}))
    }
}


export default connect(mapStateToProps,mapDispatchToProps)(AddExpenses)

