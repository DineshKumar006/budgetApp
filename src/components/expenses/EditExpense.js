import React from 'react';
import {connect} from 'react-redux';
import Form from '../form/form'
import * as ExpenseActions from '../../ReduxStore/actions/ExpenseActions'

const EditExpense=(props)=>{

    console.log(props)

   const editHandler=(expenseData)=>{
       props.editExpenses(props.match.params.id,expenseData)
       props.history.push('/')

    }
    return(
        <div>
            <p>Edit expenses {props.match.params.id}</p>
           <Form EditexpenseData={props.expenseData} submitExpenses={(expenseData)=>editHandler(expenseData)} />

        </div>
    )
}

const mapStateToProps=(state,props)=>{
    return{
        expenseData:state.expenseReducer.expenses.find(expense=>{
            return expense.id===props.match.params.id
        })
    }
}
const mapDispatchToProps=(dispatch)=>{
    return{
        editExpenses:(id,data)=>dispatch(ExpenseActions.editExpenses(id,data))
    }
}

export default connect(mapStateToProps,mapDispatchToProps)(EditExpense)