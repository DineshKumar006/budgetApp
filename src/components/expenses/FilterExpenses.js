import React,{Component} from 'react';
import * as FilterActions from '../../ReduxStore/actions/FilterActions'
import * as ExpensesActions from '../../ReduxStore/actions/ExpenseActions'
import {connect} from 'react-redux'
import {DateRangePicker} from 'react-dates'
import 'react-dates/lib/css/_datepicker.css';
import 'react-dates/initialize';



class FilterExpenses extends Component{

    constructor(){
        super()
        this.state={
            // startDate:moment(),
            // endDate:moment(),
            calenderFocused:null,
            
        }
    }

    onFocusChange=(calenderFocused)=>{
        this.setState(()=>({calenderFocused:calenderFocused}))
    }
    onDatesChange=({startDate,endDate})=>{
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }
     changeHandler=(e)=>{

        const value=e.target.value;
        if(this.props.expensesDataFromDB.expenses.length>0){
            this.props.filterByName(value.toLowerCase())   
        }
    }
     amountChangeHandler=(e)=>{
        const amount= e.target.value;
        if(this.props.expensesDataFromDB.expenses.length>0){
            this.props.filterByAmount(parseInt(amount))
        }
    }
    render(){
        




const sortHandler=(event)=>{
    //  console.log(event.target.value)
    if(this.props.expensesData.expenses.length>0 || this.props.expensesDataFromDB.expenses.length>0){
        if(event.target.value==='amount'){
            this.props.sortByAmount()
        }else if(event.target.value==='date'){
            this.props.sortByDate()
        }

    }

}

console.log(this.props.amountFocus,this.props.textFocus)
        return(
            <div>
               
                <label>FilterExpenses</label>
               
                <input type='number' placeholder='Amountsearch'  autoFocus={this.props.amountFocus}
                 onChange={this.amountChangeHandler} value={this.props.filterData.amountSearch.toString()}/>

                <input type='text' placeholder="Enter Filter" autoFocus={this.props.textFocus}
                 onChange={this.changeHandler}  value={this.props.filterData.text}/>

                <DateRangePicker
                startDate={this.props.filterData.startDate}
                startDateId="1"
                endDate={this.props.filterData.endDate}
                endDateId="2"
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.calenderFocused}
                onFocusChange={this.onFocusChange}
                numberOfMonths={1} //limit the calander view to one
                isOutsideRange={()=>false} //we  can go in past
                showClearDates={true}

                    />


<div>
    <select onChange={sortHandler} value={this.props.filterData.sortBy}>
        <option value="date">date</option>
        <option value="amount">amount</option>
    </select>

</div>

            <h4>Filter data</h4>
            
            </div>
        )
    }
}


const mapStateToProps=(state)=>{
    return{
        expensesData:state.expenseReducer,
        filterData:state.filterReducer,
        expensesDataFromDB:state.fetchExpensesReducer
    }

}

const mapDispatchToProps=(dispatch)=>{
    return{
        filterByName:(value)=>dispatch(FilterActions.filterByName(value)),
        sortByAmount:()=>dispatch(FilterActions.sortByAmount()),
        sortByDate:()=>dispatch(FilterActions.sortByDate()),
        removeExpenses:(id)=>dispatch(ExpensesActions.removeExpenses(id)),
        filterByAmount:(amount)=>dispatch(FilterActions.filterByAmount(amount)),
        setStartDate:(startDate)=>dispatch(FilterActions.setStartDate(startDate)),
        setEndDate:(endDate)=>dispatch(FilterActions.setEndDate(endDate))

    }
}


export default connect(mapStateToProps,mapDispatchToProps)(FilterExpenses)

