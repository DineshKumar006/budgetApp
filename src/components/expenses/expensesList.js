import React from 'react';
import {connect} from 'react-redux'
import * as ExpenseActions from '../../ReduxStore/actions/ExpenseActions'
import {visibleData} from '../../selector/visibleData'
import moment from 'moment';
import {Link} from 'react-router-dom'
import numeral from 'numeral'

export const ExpenseList =(props)=>{
console.log(props.expensesData.length)

let data=null,Total=0

if(props.expensesData.length>0){
    // console.log(visibleData(props.expensesData,props.filterData))
    data=visibleData(props.expensesData,props.filterData)

    data.map(ele=>{
        return Total=Total+ele.amount
    })
}

    return(
    <div>
        <hr></hr>

        <ul>
            {data!==null? <div>
            <h4>viewing {data.length} expenses. Total ${Total}</h4>
                {data.map(curr=>{
                return (<li key={curr.id}>
                    
                  <Link to={`/edit/${curr.id}`}> {curr.description} </Link> 
                  {numeral(curr.amount).format('$0,0.00')} - {moment(curr.createdAt).format('MMMM Do, YYYY')}
                    <button onClick={()=>props.removeExpenses(curr.id)}>Delete</button>

                </li>)
                })}
            </div>:'No Expenses added yet'}
        </ul>
    </div>
)
        }

const mapStateToProps=(state)=>{
    return{
        // expensesData1:state.expenseReducer,
        // filterData:state.filterReducer
    }
}

const mapDispatchToProps=(dispatch)=>{
    return{
        removeExpenses:(id)=>dispatch(ExpenseActions.removeExpenses(id))

    }
}


export default  connect(mapStateToProps,mapDispatchToProps)(ExpenseList);