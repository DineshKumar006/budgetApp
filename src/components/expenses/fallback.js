import React from 'react';

const fallback=()=><div>Loading....</div>

export default fallback;