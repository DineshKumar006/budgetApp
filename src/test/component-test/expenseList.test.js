import React from 'react';
import {shallow} from 'enzyme'
import toJSON from 'enzyme-to-json';
import {ExpenseList} from '../../components/expenses/expensesList'
 import expenses from '../fixtures/expenses'

 const filterData={
    text:'',
    startDate: undefined,
    endDate:undefined,
    sortBy:'date', 
    amountSearch:NaN   
}


test('expense List test validation',()=>{
    console.log(expenses.length)
    const wrapper=shallow(<ExpenseList expensesData={expenses} filterData={filterData}/>)

    expect(toJSON(wrapper)).toMatchSnapshot()

});


test('expense List for empty data test validation',()=>{
    console.log(expenses.length)
    const wrapper=shallow(<ExpenseList expensesData={[]} filterData={filterData}/>)

    expect(toJSON(wrapper)).toMatchSnapshot()

})