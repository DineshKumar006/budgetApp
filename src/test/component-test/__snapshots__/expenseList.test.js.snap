// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expense List for empty data test validation 1`] = `
<div>
  <hr />
  <ul>
    No Expenses added yet
  </ul>
</div>
`;

exports[`expense List test validation 1`] = `
<div>
  <hr />
  <ul>
    <div>
      <h4>
        viewing 
        3
         expenses. Total $
        2300
      </h4>
      <li
        key="3"
      >
        <Link
          to="/edit/3"
        >
           
          curd
           
        </Link>
          -
        1000
         - 
        345600000
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
      <li
        key="1"
      >
        <Link
          to="/edit/1"
        >
           
          rice
           
        </Link>
          -
        1200
         - 
        0
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
      <li
        key="2"
      >
        <Link
          to="/edit/2"
        >
           
          milk
           
        </Link>
          -
        100
         - 
        -345600000
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </li>
    </div>
  </ul>
</div>
`;
